### ISSUE TITLE
There should be an MSBuildLocator.Dispose() method

### ISSUE THREAD

**Issue by DustinCampbell** [Microsoft/MSBuildLocator#5]
Currently, MSBuildLocator hooks the AppDomain.AssemblyResolve event and that hook will stay in place for the lifetime of the AppDomain. However, in some scenarios, it is necessary to unhook and this event if several MSBuildLocators might be constructed (for example, in a unit testing scenario).

**Comment by AndyGerlicher** [comment]
Makes sense. I thought about that but didn't get around to it (methods are all static so not quite that easy). One issue though is by the time anything is loaded it's really too late to remove that hook in a meaningful way. And it won't get called for those assemblies on that AppDomain anyway if they're already loaded.

**Comment by tibel** [comment]
For unit tests a property `IsRegistered` would help.
Subsequent calls to `RegisterInstance()` throw `InvalidOperationException`.
